extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceJdbcRuntimeConfig.java

```diff
@@ -8,100 +8,71 @@
 import io.quarkus.runtime.annotations.ConfigItem;
 
 @ConfigGroup
-public class DataSourceRuntimeConfig {
+public class DataSourceJdbcRuntimeConfig {
 
     /**
      * The datasource URL
      */
     @ConfigItem
-    public Optional<String> url;
-
-    /**
-     * The datasource username
-     */
-    @ConfigItem
-    public Optional<String> username;
-
-    /**
-     * The datasource password
-     */
-    @ConfigItem
-    public Optional<String> password;
-
-    /**
-     * The credentials provider name
-     */
-    @ConfigItem
-    public Optional<String> credentialsProvider;
-
-    /**
-     * The credentials provider type.
-     * <p>
-     * It is the {@code &#64;Named} value of the credentials provider bean. It is used to discriminate if multiple
-     * CredentialsProvider beans are available.
-     * <p>
-     * For Vault it is: vault-credentials-provider. Not necessary if there is only one credentials provider available.
-     */
-    @ConfigItem
-    public Optional<String> credentialsProviderType;
+    public Optional<String> url = Optional.empty();
 
     /**
      * The initial size of the pool. Usually you will want to set the initial size to match at least the
      * minimal size, but this is not enforced so to allow for architectures which prefer a lazy initialization
      * of the connections on boot, while being able to sustain a minimal pool size after boot.
      */
     @ConfigItem
-    public Optional<Integer> initialSize;
+    public Optional<Integer> initialSize = Optional.empty();
```

Comment:
Why is this necessary? I thought these fields would be set when injected?

Reply:
Already explained above.

Reply:
It should be an `OptionalInt` at any rate.